### PR TITLE
Show post type in post text

### LINKED ISSUE #735: Show post type in post_select field
Related to #627

I have been using `post_select` for a client and they we were querying posts from two post types. The posts in both post types had the same titles. There was no way to differentiate between the two posts.

image

[Customize Object Selector](https://github.com/xwp/wp-customize-object-selector) has a feature where it shows the post type.
image

Beitrag = Post
Inhaltselement = Content Block (CPT)

### PR DESCRIPTION
**Pull request by grappler**
fixes #735

### CODE REVIEW

File: inc/fields/class-shortcode-ui-field-post-select.php [hunk 1]
```diff
@@ -137,11 +137,21 @@ public function action_wp_ajax_shortcode_ui_post_field() {
 
 		$query = new WP_Query( $query_args );
 
+		$is_multiple_post_types = count( $query->get( 'post_type' ) ) > 1;
+
 		foreach ( $query->posts as $post_id ) {
+			$post_type     = get_post_type( $post_id );
+			$post_type_obj = get_post_type_object( $post_type);
```
Comment (szepeviktor) [inline inc/fields/class-shortcode-ui-field-post-select.php]: Please follow WPCS in this line.

File: inc/fields/class-shortcode-ui-field-post-select.php [hunk 2]
```diff
@@ -137,11 +137,21 @@ public function action_wp_ajax_shortcode_ui_post_field() {
...
+
+			$text = html_entity_decode( get_the_title( $post_id ) );
+
+			if ( $is_multiple_post_types && $post_type_obj ) {
+				$text .= sprintf( ' (%1$s)', $post_type_obj->labels->singular_name );
```
Comment (szepeviktor) [inline inc/fields/class-shortcode-ui-field-post-select.php:149]: Maybe somehow escape `singular_name`?
Although we have json encoding at the end.

File: inc/fields/class-shortcode-ui-field-post-select.php (hunk 1 above)
Comment (goldenapples) [inline inc/fields/class-shortcode-ui-field-post-select.php]: This line is triggering a code sniff violation: `Expected 1 spaces before closing bracket`.

**Review by goldenapples — COMMENTED** [review]
I like this and would support merging once if the one whitespace issue I noted is cleaned up.

It'd be good to tackle providing a better way to template the results for post selection (ie adding the featured image as suggested in #627), but that doesn't need to be done here.

**Review by goldenapples — APPROVED** [review]
:+1: Anyone else have suggestions on this? Otherwise, I can merge...

File: inc/fields/class-shortcode-ui-field-post-select.php [hunk 3]
```diff
@@ -137,11 +137,21 @@ public function action_wp_ajax_shortcode_ui_post_field() {
 
 		$query = new WP_Query( $query_args );
 
+		$is_multiple_post_types = count( $query->get( 'post_type' ) ) > 1;
```
Comment (rahulsprajapati) [inline inc/fields/class-shortcode-ui-field-post-select.php]: Maybe we should consider 'post_type' => 'any' value.

Comment (grappler) [reply inc/fields/class-shortcode-ui-field-post-select.php]: I think the `post_type`  should be left to when setting up the setting. For a certain dropdown I may only want posts and in another just pages.

The code is just counting the number of different types of post types in the query results.

Comment (rahulsprajapati) [reply inc/fields/class-shortcode-ui-field-post-select.php]: yes, but if we use 'post_type' => 'any' that means in this query we will get all post types posts. And still the `count( $query->get( 'post_type' ) )` will be 1. And at that time we won't get post type name in the select dropdown. As we are using this condition.
https://codex.wordpress.org/Class_Reference/WP_Query#Type_Parameters

Comment (grappler) [reply inc/fields/class-shortcode-ui-field-post-select.php]: Ah, I get you now. The last commit should fix that.

**Review by mattheu — APPROVED** [review]

**Review by szepeviktor — APPROVED** [review]

**Comment by grappler** [comment]
Can this be merged in? Three people have approved the PR.

**Comment by mattheu** [comment]
👍

**Pull request merged**